### PR TITLE
Fix bugs caused by bad adherence to rules

### PR DESCRIPTION
**Pull request by WillCWX**
# Pull Request

## Description

I made some mistakes when trying to adhere to the new rules, this caused some bugs

- `() => void(async () => ...)` doesn't work at all
  - Fix: Changed to `() => void asyncFunction()` which i can confirm works
- To make ensure `no_unused_var` was adhered to, i left `_response = await api` in `api-wrapper.ts`, this caused the frontend to crash when called as there is no error handling.
  - Fix: Added error handling 

## Checklist

- [X] I have checked that the changes included in the PR are intended to merge to `master` or any destination branch.
- [X] I have verified that the new changes do not break any existing functionalities, unless the new changes are intended and have approved by the team.
- [X] I will take care of the merging and delete the side-branch after the PR is merged.

### CODE REVIEW

File: frontend/src/helpers/assignment-service/api-wrapper.ts [hunk 1]
```diff
@@ -14,7 +14,9 @@ const getAssignmentById = async ({
 }: {
   assignmentId: string;
 }) => {
-  const _response = await api.get(`/assignment/${assignmentId}`);
+  const _response = await api
```
Comment (tryyang2001) [inline frontend/src/helpers/assignment-service/api-wrapper.ts:17]: What will happen if we comment out this code? As we don't need to make the API call for now, I don't think it makes sense to keep this code here?

Comment (WillCWX) [reply frontend/src/helpers/assignment-service/api-wrapper.ts:17]: you get unused -vars errors with api and the imports

Comment (tryyang2001) [reply frontend/src/helpers/assignment-service/api-wrapper.ts:17]: Oh, then I think for now we can remove the import statement first?

Comment (WillCWX) [reply frontend/src/helpers/assignment-service/api-wrapper.ts:17]: nah leave it like this, removing the `api` means removing `await` which means the function cannot be `async` which breaks a lot of your other stuff

**Pull request merged**